### DISCUSSION TITLE
howto  record the method return value?

### DISCUSSION THREAD

**Discussion by szhengli** [apache/skywalking#9784] [General]
Does skywalking support recordinf  the return values of a method invocation? if so, how?   thanks

**Comment by pg-yang** [comment]
You could  refer to  https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/java-plugin-development-guide/  to develop own plugins . It's easy to collect  method return value at `InstanceMethodsAroundInterceptor#afterMethod` method

**Comment by wu-sheng** [comment]
Another option is through returnedObj in the Tag annotation

https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/application-toolkit-trace/
